pkg/strategy/blocking_strategy.go

```diff
@@ -0,0 +1,44 @@
+package strategy
+
+import (
+	"fmt"
+	"log"
+	"net/http"
+	"time"
+
+	"github.com/acouvreur/traefik-ondemand-plugin/pkg/pages"
+)
+
+type BlockingStrategy struct {
+	Request    string
+	Name       string
+	Next       http.Handler
+	Timeout    time.Duration
+	BlockDelay time.Duration
+}
+
+// ServeHTTP retrieve the service status
+func (e *BlockingStrategy) ServeHTTP(rw http.ResponseWriter, req *http.Request) {
+
+	// TODO: Wait for 
+	for start := time.Now(); time.Since(start) < e.BlockDelay; {
```

Comment:
I tested it some more! It would appear this code will hammer the on demand service, making traefik and the service hang. I tried to come up with some back-off system, but since I'm no where proficient in Go, I only could come up with adding `time.Sleep(1 * time.Second)` at the end of the for loop. No idea if that has implications for blocking the thread and maybe it blocks other behavior. Otherwise a non blocking backoff system would be better to schedule retrying and keep the thread free for other work. Or is this not something to worry about?

Reply:
It indeed needs to wait before retrying, otherwise it will hammer the service as you described.
A simple wait should not block any other request.
Each request is handled by a goroutine, thus not blocking any other work from the app.